File: mobile-client/android/UseCase/Message/Thread/SendMessageUseCase.kt
package app.mobiling.client.client.usecase.message.thread

import app.mobiling.client.client.contract.message.thread.MessageItemPayload
import app.mobiling.client.client.contract.message.thread.SendMessageRequest
import app.mobiling.client.client.data.message.thread.MessageThreadGateway

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
class SendMessageUseCase(
    private val gateway: MessageThreadGateway,
) {
    suspend operator fun invoke(request: SendMessageRequest): MessageItemPayload = gateway.sendMessage(request)
}

File: client/ios/UseCase/Message/Thread/SendMessageUseCase.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
public struct SendMessageUseCase: Sendable {
    private let gateway: MessageThreadGateway

    public init(gateway: MessageThreadGateway) {
        self.gateway = gateway
    }

    public func callAsFunction(request: SendMessageRequest) async throws -> MessageItemPayload {
        try await gateway.sendMessage(request: request)
    }
}
